taskcluster/app_services_taskgraph/transforms/worker.py

```diff
@@ -0,0 +1,25 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from taskgraph.transforms.base import TransformSequence
+
+transforms = TransformSequence()
+
+@transforms.add
+def setup_worker(_, tasks):
+    for task in tasks:
+        task_name = task['name']
+        try:
+            worker_type = task['worker-type']
+        except KeyError:
+            raise ValueError(f"worker-type not set for {task_name}")
+        if worker_type == 'linux':
+            task['worker-type'] = 'b-linux'
+            worker = task.setdefault('worker', {})
+            worker['docker-image'] = {'in-tree': 'linux'}
+        elif worker_type == 'osx':
+            task['worker-type'] = 'b-osx'
```

Comment:
I think I would avoid translating the `worker-type` here as it just adds another layer of indirection. If you want to get rid of the `b-` prefix, you can rename the aliases here:
https://github.com/mozilla/application-services/blob/main/taskcluster/ci/config.yml#L17

You could continue setting up the `docker-image` here if you like however.